### ISSUE TITLE
Regex Problems with database schema and table name

### ISSUE THREAD

**Issue by Duvenbeck-SWE** [sqlfluff/sqlfluff#5481]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I tried to format the following SQL-Query.
```sql
select
    'Tabellen' as Objekt,
    Count(*) as Anzahl
from dbo.sql_modules
```
The result was that no fixable linting violations were found but 1 templating/parsing error.
Therefore, I tried parsing the Query and got the following result:
```
==== parsing violations ====
L:   4 | P:   1 |  PRS | Line 4, Position 1: Found unparsable section: 'from
                       | dbo.sql_modules'
```
In the parsing result it says:
```
[L:  4, P:  1]      |    unparsable:                                               !! Expected: 'Nothing else in FileSegment.'
[L:  4, P:  1]      |        word:                                                 'from'
[L:  4, P:  5]      |        whitespace:                                           ' '
[L:  4, P:  6]      |        unquoted_relative_sql_file_path:                      'dbo.sql'
[L:  4, P: 13]      |        word:                                                 '_modules'
```

### Expected Behaviour

I would expect sqlfluff to recognize the dbo.sql_module as a schema followed by the table name.

### Observed Behaviour

But it does not.
So I looked into the problem and I think I found the responsible Regex pattern.
In the file `sqlfluff-main\src\sqlfluff\dialects\dialect_tsql.py` in line 180 you can find the following code snippet
```py
RegexLexer(
           "unquoted_relative_sql_file_path",
           # currently there is no way to pass `regex.IGNORECASE` flag to `RegexLexer`
           r"[.\w\\/#-]+\.[sS][qQ][lL]",
           CodeSegment,
       ),
```
The Regex pattern `[.\w\\/#-]+\.[sS][qQ][lL]` forbids the use of 'sql' after the dot. If you put anything else it works fine (e.g. 'wql').
But what to do if your table name starts with sql as is the case in my example above ('sql_module').

### How to reproduce

The problem can be reproduced as described above.

### Dialect

- T-SQL

### Version

- 2.3.4
- but the above mentioned Regex-Pattern can be found in the latest GitHub Repository too (as of now: 2.3.5)

### Configuration

- the default configuration file can be used to recreate

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
